src/libraries/tests.proj

```diff
@@ -320,10 +320,15 @@
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Threading.Timer/tests/System.Threading.Timer.Tests.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="$(MSBuildThisFileDirectory)System.Collections\tests\System.Collections.Tests.csproj"
+                      Condition="'$(TestAssemblies)' == 'true' and '$(TestSingleFile)' == 'true'" />
+  </ItemGroup>
+
   <ItemGroup>
     <ProjectReference Include="$(MSBuildThisFileDirectory)*\tests\**\*.Tests.csproj"
                       Exclude="@(ProjectExclusions)"
-                      Condition="'$(TestAssemblies)' == 'true'" />
+                      Condition="'$(TestAssemblies)' == 'true' and '$(TestSingleFile)' != 'true'" />
```

Comment:
The pattern in this file is to add to ProjectExclustions instead of modifying this part. Something like:

```xml
  <ItemGroup Condition="'$(TestSingleFile)' == 'true'">
    <!-- Run only a small randomly chosen set of passing test suites -->
    <ProjectExclusions Include="$(MSBuildThisFileDirectory)*\tests\**\*.Tests.csproj" />
    <ProjectExclusions Remove="$(MSBuildThisFileDirectory)System.Collections\tests\System.Collections.Tests.csproj" />
  </ItemGroup>
```

should work.